index.ts

```diff
@@ -269,85 +271,94 @@ export default class Analytics {
   /**
    * Flushes the message queue to the server immediately if a flush is not already in progress.
    */
-  flush(callback: AnalyticsFlushCallback = () => {}): void {
-    // check if earlier flush was pushed to queue
-    this.logger.debug('in flush');
-
-    if (!this.enable) {
-      setImmediate(callback);
-      return;
-    }
-
-    if (this.timer) {
-      this.logger.debug('cancelling existing timer...');
-      clearTimeout(this.timer);
-      this.timer = null;
-    }
-
-    if (!this.queue.length) {
-      this.logger.debug('queue is empty, nothing to flush');
-      setImmediate(callback);
-      return;
-    }
+  async flush(callback: AnalyticsFlushCallback = () => {}): Promise<FlushResponse | void> {
+    return await new Promise(async (resolve, reject) => {
```

Comment:
the tests expect a promise rejection if the request has failed (which is why we were able to remove the `error` property from `FlushResponse`). I personally think that rejecting the promise in the case of a failure is not the behavior we want because this will create a `console.error` message. We'd just like the client to take care of this situation on its own and build handling for it, instead of alerting the user.

I reworked the PR so flush returns an error object instead of throwing in these cases.